src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -154,6 +154,93 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
             }
         }
 
+        /// <summary>
+        /// Adds the given <paramref name="classValue"/> to the <paramref name="tagHelperOutput"/>'s
+        /// <see cref="TagHelperOutput.Attributes"/>.
+        /// </summary>
+        /// <param name="tagHelperOutput">The <see cref="TagHelperOutput"/> this method extends.</param>
+        /// <param name="classValue">The class value to add.</param>
+        public static void AddClass(
+            this TagHelperOutput tagHelperOutput,
+            string classValue)
+        {
+            if (tagHelperOutput == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelperOutput));
+            }
+
+            if (!tagHelperOutput.Attributes.ContainsName("class"))
+            {
+                tagHelperOutput.Attributes.Add("class", classValue);
+            }
+            else
+            {
+                var classAttributeValue = tagHelperOutput.Attributes["class"].Value.ToString();
```

Comment:
Definitely 😄, so it can have `IHtmlContent`, `string`s, `non-strings` and `HtmlString`s. For example, this is how Damian handled the various value types: https://github.com/aspnet/live.asp.net/blob/d8d0958aabea324f48e4ffc50b5a38899ff4018b/src/live.asp.net/TagHelpers/ScriptInliningTagHelper.cs#L38-L64